html5validator/validator.py

```diff
@@ -144,14 +157,24 @@ def validate(self, files):
 
         e = stderr.splitlines()
 
+        # Removes any empty items in the list
+        e = list(filter(None, e))
+
+        # Prevents removal of xml tags if there are errors
+        if self.format == "xml" and len(e) > 4:
+            self.ignore = self.ignore[3:]
+
+        LOGGER.debug(e)
+
         for i in self.ignore:
             e = [l for l in e if i not in l]
         for i in self.ignore_re:
             regex = re.compile(i)
             e = [l for l in e if not regex.search(l)]
 
-        if stderr:
-            LOGGER.error(stderr)
+        if e:
+            LOGGER.error(e)
+            return len(e)
         else:
             LOGGER.info('All good.')
-        return len(e)
+            return len(e)
```

Comment:
Was this change intentional?

Reply:
It was but I fixed it now.
Before both the if and else had their own return but I moved it back so there is only one.